### PR TITLE
Added KL divergence calculation between two Cauchy distributions.

### PR DESCRIPTION
**Pull request by gergely-flamich**
KL calculation is based on [this paper](https://arxiv.org/pdf/1905.10965.pdf).

### CODE REVIEW

File: tensorflow_probability/python/distributions/cauchy_test.py [hunk 1]
```diff
@@ -432,6 +432,33 @@ def testAssertsPositiveScaleAfterMutation(self):
...
+    c_b = tfd.Cauchy(loc=mu_b, scale=gamma_b, validate_args=True)
+
+    kl = tfd.kl_divergence(c_a, c_b)
+    reverse_kl = tfd.kl_divergence(c_b, c_a)
+    kl_val = self.evaluate(kl)
+    reverse_kl_val = self.evaluate(reverse_kl)
+
+    kl_expected = np.log((gamma_a + gamma_b)**2 + (mu_a - mu_b)**2) - np.log(4. * gamma_a * gamma_b)
+
+    x = c_a.sample(int(1e5), seed=test_util.test_seed())
```
Comment (srvasude) [inline tensorflow_probability/python/distributions/cauchy_test.py:452]: Let's split these tests up (testing against numpy vs. sampling based test.)

Comment (gergely-flamich) [reply tensorflow_probability/python/distributions/cauchy_test.py:479]: I split the test into two functions, one with the "WithAnalytic" and other with the "WithMC" postfix, and moved the numpy and sampling-based tests to them respectively.

Also note: I based the tolerances for the MC tests on the ones for tfd.Normal

Comment (gergely-flamich) [reply tensorflow_probability/python/distributions/cauchy_test.py:479]: Oh also, the list of registered KLs is reported here:
https://www.tensorflow.org/probability/api_docs/python/tfp/distributions/kl_divergence

Should I also update this, or is this done automatically / internally after the merge?

Comment (srvasude) [reply tensorflow_probability/python/distributions/cauchy_test.py:479]: This will be done automatically.

**Pull request merged**